contracts/Compass.vy

```diff
@@ -250,3 +317,107 @@ def deploy_erc20(_paloma_denom: String[64], _name: String[64], _symbol: String[3
     event_id: uint256 = unsafe_add(self.last_event_id, 1)
     self.last_event_id = event_id
     log ERC20DeployedEvent(_paloma_denom, erc20, _name, _symbol, _decimals, event_id)
+
+@external
+@view
+def arbitrary_view(contract_address: address, payload: Bytes[1024]) -> Bytes[1024]:
+    return raw_call(contract_address, payload, is_static_call=True, max_outsize=1024)
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
+# F E E S
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
+
+# Deposit some balance on the contract to be used when sending messages from Paloma.
+# depositor_paloma_address: paloma address to which to attribute the sent amount
+# amount: amount of COIN to register with compass. Overpaid balance will be sent back.
+@external
+@payable
+@nonreentrant('lock')
+def deposit(depositor_paloma_address: bytes32, amount: uint256):
+    if amount > msg.value:
+        send(msg.sender, unsafe_sub(amount, msg.value))
+    else:
+        assert amount == msg.value, "Insufficient deposit"
```

Comment:
@webelf101 

This seems to be buggy. If I set `amount` to `100` or so, but in reality I only send `1`, compass is trying to send back 99.